src/Control/Distributed/Process/Tests/CH.hs

```diff
@@ -1316,6 +1321,53 @@ testUnsafeSendChan TestTransport{..} = do
 
   takeMVar clientDone
 
+testCallLocal :: TestTransport -> Assertion
+testCallLocal TestTransport{..} = do
+  node <- newLocalNode testTransport initRemoteTable
+
+  -- Testing that (/=) <$> getSelfPid <*> callLocal getSelfPid.
+  result <- newEmptyMVar
+  runProcess node $ do
+    r <- (/=) <$> getSelfPid <*> callLocal getSelfPid
+    liftIO $ putMVar result r
+  True <- takeMVar result
+  return ()
+
+  -- Testing that when callLocal is interrupted, the worker is interrupted.
+  ibox <- newIORef False
+  runProcess node $ do
+    keeper <- getSelfPid
+    spawnLocal $ do
+        caller <- getSelfPid
+        send keeper caller
+        onException 
+          (callLocal $ do
+                onException (do send keeper ()
+                                expect)
+                            (do liftIO $ writeIORef ibox True))
+          (send keeper ())
+    caller <- expect
```

Comment:
This `expect` call is unnecessary if the `caller` is sent by the worker instead of `()`.